tests/phpunit/tests/image/meta.php

```diff
@@ -6,9 +6,28 @@
  * @group upload
  * @requires extension gd
  * @requires extension exif
+ *
+ * @covers ::wp_read_image_metadata
  */
 class Tests_Image_Meta extends WP_UnitTestCase {
 
+	public static function setUpBeforeClass() {
+		if ( ! in_array( 'wptest', stream_get_wrappers(), true ) ) {
+			require_once DIR_TESTROOT . '/includes/class-wp-test-stream.php';
+			stream_wrapper_register( 'wptest', 'WP_Test_Stream' );
+		}
+
+		WP_Test_Stream::$data = array(
+			'wp_read_image_metadata' => array(
+				'/image1.jpg' => file_get_contents( DIR_TESTDATA . '/images/test-image-upside-down.jpg' ),
+				'/image2.jpg' => file_get_contents( DIR_TESTDATA . '/images/2004-07-22-DSC_0007.jpg' ),
+				'/image3.jpg' => file_get_contents( DIR_TESTDATA . '/images/33772.jpg' ),
+			),
+		);
+
+		parent::setUpBeforeClass();
+	}
```

Comment:
```suggestion
	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
		if ( ! in_array( 'wptest', stream_get_wrappers(), true ) ) {
			require_once DIR_TESTROOT . '/includes/class-wp-test-stream.php';
			stream_wrapper_register( 'wptest', 'WP_Test_Stream' );
		}

		WP_Test_Stream::$data = array(
			'wp_read_image_metadata' => array(
				'/image1.jpg' => file_get_contents( DIR_TESTDATA . '/images/test-image-upside-down.jpg' ),
				'/image2.jpg' => file_get_contents( DIR_TESTDATA . '/images/2004-07-22-DSC_0007.jpg' ),
				'/image3.jpg' => file_get_contents( DIR_TESTDATA . '/images/33772.jpg' ),
			),
		);
	}
```

It might be worth using a unique stream name to avoid the clash with `Tests_Image_Editor_Imagick` and calling `stream_wrapper_unregister()` in `wpTearDownAfterClass()`. But that's a real nitpick, this looks good once the function name is change to `wpSetUpBeforeClass()`

Reply:
>Super minor change. The diff in the suggestion is not helpful, it's to use wpSetUpBeforeClass() rather than setUpBeforeClass().

@peterwilsoncc The factory instance is not being used in the method. I'm curious: why use `wpSetUpBeforeClass()` rather than `setUpBeforeClass()`? 

>It might be worth using a unique stream name to avoid the clash with Tests_Image_Editor_Imagick and calling stream_wrapper_unregister() in wpTearDownAfterClass(). But that's a real nitpick, this looks good once the function name is change to wpSetUpBeforeClass()

100% agree with you.

Reply:
>It might be worth using a unique stream name to avoid the clash with Tests_Image_Editor_Imagick and calling stream_wrapper_unregister() in wpTearDownAfterClass(). But that's a real nitpick, this looks good once the function name is change to wpSetUpBeforeClass()

Commit https://github.com/WordPress/wordpress-develop/pull/1131/commits/9fc13f9d0444aef1cb61bcbbae56cf05389e5472 unregisters

Commit https://github.com/WordPress/wordpress-develop/pull/1131/commits/abce31bc7455c699ac6995be9c624503332f292f changes to a unique stream name

Great suggestions @peterwilsoncc!

Reply:
> I'm curious: why use `wpSetUpBeforeClass()` rather than `setUpBeforeClass()`?

`wpSetUpBeforeClass()` is generally preferred in the test suite as it's called automatically if it exists without relying on the need to call the parent class. I don't know why there is an inconsistency with `setUp()` either :D  I included the factory in the suggestion as it appears to be defined elsewhere regardless of whether it's used or not.

Reply:
For anyone following along with this discussion, Peter and I had a follow-up conversation. Results:

For all instances in Core, use `wpSetUpBeforeClass()` instead of `setUpBeforeClass()`.

Reply:
@peterwilsoncc switched both to use the `wp` versions in commit https://github.com/WordPress/wordpress-develop/pull/1131/commits/066b88c7bd034c85b0dc71b15512d553cdf7d477.